lib/player.js

```diff
@@ -1698,6 +1698,22 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
           await this.srcEqualsInner_(startTimeOfLoad, mimeType);
         }, 'srcEqualsInner_');
       } else {
+        // Wait for the manifest to be parsed.
+        await mutexWrapOperation(async () => {
+          // Also wait for finish, so this stops on errors.
+          const promises = [
+            preloadManager.waitForFinish(),
+            preloadManager.waitForManifest(),
+          ];
+          await Promise.race(promises);
```

Comment:
I wonder couldn't we reject manifest promise when an error occurs or is it not enough?

Reply:
I was doing that at first, but that was what was causing the test failures, I believe.

Reply:
Maybe this race should be move to `waitForManifest()` method then?

Reply:
I tend to agree.  Let's move it, please.

Reply:
Okay, trying that out.